dla_cnn/desi/preprocess.py

```diff
@@ -92,3 +92,33 @@ def label_sightline(sightline, kernel, REST_RANGE, pos_sample_kernel_percent=0.3
     # classification is 1 / 0 / -1 for DLA/nonDLA/border
     # offsets_array is offset
     return classification, offsets_array, column_density
+
+def _normalize(sightline, camera):
+    '''
+    Normalize spectrum by dividing the mean value of continnum at lambda[left,right]
+    ------------------------------------------
+    parameters:
+    
+    sightline: dla_cnn.data_model.Sightline.Sightline object;
+    camera : str, 'b' : the blue channel of the specctra, 'r': the r channel of the spectra,
+                  'z' : the z channel of the spectra, 'all': all spectra
+    
+    --------------------------------------------
+    return
+    
+    sightline: the sightline after normalized
+    
+    '''
+    def _normalize(sightline,left,right):
+        rest_wavelength = (10**sightline.loglam)/(1+sightline.z_qso)
+        test =(rest_wavelength>=left)&(rest_wavelength<=right)
+        sightline.flux = sightline.flux/np.median(sightline.flux[test])
+        
+    if camera == 'b':
```

Comment:
This isn't going to work well in general.

Here are two different options to consider:

1.  Find which camera contains 1450A in the rest frame.  And use it to normalize all cameras.
2.  Find the camera with the most coverage of the Lya forest (1070-1170) and use that instead.

I somewhat prefer the 2nd option.